vendor/github.com/ethereum/go-ethereum/rpc/client.go

```diff
@@ -421,7 +420,15 @@ func (c *Client) send(ctx context.Context, op *requestOp, msg interface{}) error
 		log.Trace("", "msg", log.Lazy{Fn: func() string {
 			return fmt.Sprint("sending ", msg)
 		}})
+
+		retries := 0
 		err := c.write(ctx, msg)
+		for err != nil && retries < failedWriteRetries {
```

Comment:
If a call to `write` fails, `c.writeConn` will be set to nil. Then, on the next call the RPC client will retry to reestablish the connection https://github.com/livepeer/go-livepeer/blob/e67b8370df00d5f8bf8a05d16d81a2c8648c7866/vendor/github.com/ethereum/go-ethereum/rpc/client.go#L450